experimenter/experimenter/segments/tests/test_segments.py

```diff
@@ -0,0 +1,90 @@
+
+
+@mock_valid_segments
+class TestSegments(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        Segments.clear_cache()
+
+    def test_load_all_segments(self):
+        segments = Segments.all()
+        self.assertEqual(len(segments), 4)
+        self.assertIn(
+            Segment(
+                slug="fenix_segment",
+                friendly_name="Fenix Segment",
+                application=NimbusExperiment.Application.FENIX,
+                description="Fenix segment for testing",
+                select_expression="{{agg_sum('ad_click')}}",
+            ),
+            segments,
+        )
+        for i in range(1, 4):
+            self.assertIn(
+                Segment(
+                    slug=f"desktop_segment_{i}",
+                    friendly_name=f"Desktop Segment {i}",
+                    application=NimbusExperiment.Application.DESKTOP,
+                    description="Firefox desktop segment used for testing",
+                    select_expression="country_code in ('IN', 'US')",
+                ),
+                segments,
+            )
+
+    def test_load_segments_by_application(self):
+        fenix_segments = Segments.by_application(NimbusExperiment.Application.FENIX)
+        self.assertEqual(len(fenix_segments), 1)
+        self.assertIn(
+            Segment(
+                slug="fenix_segment",
+                friendly_name="Fenix Segment",
+                application=NimbusExperiment.Application.FENIX,
+                description="Fenix segment for testing",
+                select_expression="{{agg_sum('ad_click')}}",
+            ),
+            fenix_segments,
+        )
+
+        desktop_segments = Segments.by_application(NimbusExperiment.Application.DESKTOP)
+        self.assertEqual(len(desktop_segments), 3)
+        for i in range(1, 4):
+            self.assertIn(
+                Segment(
+                    slug=f"desktop_segment_{i}",
+                    friendly_name=f"Desktop Segment {i}",
+                    application=NimbusExperiment.Application.DESKTOP,
+                    description="Firefox desktop segment used for testing",
+                    select_expression="country_code in ('IN', 'US')",
+                ),
+                desktop_segments,
+            )
+
+
+@mock_invalid_segments
+class TestInvalidSegments(TestCase):
+    def setUp(self):
+        Segments.clear_cache()
+
+    def test_invalid_segments_do_trigger_check_error(self):
+        errors = check_segment_tomls(None)
+        self.assertEqual(
+            errors,
+            [Error(msg="Error loading Segment TOMLS: 'invalid_app_name'")],
+        )
+
+
+@mock_valid_segments
+class TestCheckSegmentTOMLs(TestCase):
+    def setUp(self):
+        Segments.clear_cache()
+
+    def test_valid_segments_do_not_trigger_check_error(self):
+        errors = check_segment_tomls(None)
+        self.assertEqual(errors, [])
```

Comment:
Consider combining these into a single class with two functions. The setup is the same for each so splitting them isn't necessary (and duplicates code), and I'd say the name `TestCheckSegmentTOMLs` describes the behavior of both tests pretty well. You can move the decorator to the functions instead of the classes.